### PR TITLE
Feature/clang tidy

### PR DESCRIPTION
**Pull request by JonasGilg**

### CODE REVIEW

File: src/DipStrikeTool.hpp [hunk 1]
```diff
@@ -60,7 +67,6 @@ class DipStrikeTool : public IVistaOpenGLDraw, public cs::core::tools::MultiPoin
   glm::dvec4 getInterpolatedPosBetweenTwoMarks(cs::core::tools::DeletableMark const& pMark1,
       cs::core::tools::DeletableMark const& pMark2, double value);
 
- private:
```
Comment (Schneegans) [inline src/DipStrikeTool.hpp:63]: Why has this been removed?

File: src/DipStrikeTool.hpp [hunk 2]
```diff
@@ -79,15 +85,13 @@ class DipStrikeTool : public IVistaOpenGLDraw, public cs::core::tools::MultiPoin
...
   glm::vec3 mNormal = glm::vec3(0.0), mMip = glm::vec3(0.0);
-  float     mOffset, mSizeFactor = 1.5f, mOpacity = 0.5;
+  float     mOffset{}, mSizeFactor = 1.5F, mOpacity = 0.5;
 
   int mScaleConnection = -1;
 
-  static const int         RESOLUTION;
-  static const std::string SHADER_VERT;
-  static const std::string SHADER_FRAG;
+  static const int RESOLUTION;
```
Comment (Schneegans) [inline src/DipStrikeTool.hpp]: Why is RESOLUTION still a class member while the others have been moved to the source file?

File: src/PathTool.hpp [hunk 3]
```diff
@@ -55,10 +62,9 @@ class PathTool : public IVistaOpenGLDraw, public cs::core::tools::MultiPointTool
 
   /// Returns the interpolated position in cartesian coordinates. The fourth component is height
   /// above the surface.
-  glm::dvec4 getInterpolatedPosBetweenTwoMarks(cs::core::tools::DeletableMark const& pMark1,
-      cs::core::tools::DeletableMark const& pMark2, double value, double const& scale);
+  glm::dvec4 getInterpolatedPosBetweenTwoMarks(cs::core::tools::DeletableMark const& l0,
+      cs::core::tools::DeletableMark const& l1, double value, double const& scale);
 
- private:
```
Comment (Schneegans) [inline src/PathTool.hpp:61]: Why removed?

File: src/PathTool.hpp [hunk 4]
```diff
@@ -85,8 +91,8 @@ class PathTool : public IVistaOpenGLDraw, public cs::core::tools::MultiPointTool
 
   int mNumSamples = 256;
 
-  static const std::string SHADER_VERT;
-  static const std::string SHADER_FRAG;
+  static const char* SHADER_VERT;
+  static const char* SHADER_FRAG;
```
Comment (Schneegans) [inline src/PathTool.hpp:95]: In the other class the static members got removed.

File: src/PolygonTool.hpp [hunk 5]
```diff
@@ -86,11 +93,10 @@ class PolygonTool : public IVistaOpenGLDraw, public cs::core::tools::MultiPointT
   // Checks if point is inside of the polygon or not
   bool checkPoint(glm::dvec2 const& point);
 
- private:
```
Comment (Schneegans) [inline src/PolygonTool.hpp:89]: Why removed?

File: src/PolygonTool.hpp [hunk 6]
```diff
@@ -124,23 +130,23 @@ class PolygonTool : public IVistaOpenGLDraw, public cs::core::tools::MultiPointT
...
+  double     mOffset{};
   glm::dvec3 mNormal2      = glm::dvec3(0.0);
   glm::dvec3 mMiddlePoint2 = glm::dvec3(0.0);
 
-  static const int         NUM_SAMPLES;
-  static const std::string SHADER_VERT;
-  static const std::string SHADER_FRAG;
+  static const int   NUM_SAMPLES;
+  static const char* SHADER_VERT;
+  static const char* SHADER_FRAG;
```
Comment (Schneegans) [inline src/PolygonTool.hpp:149]: Still class members?

**Review by Schneegans — CHANGES_REQUESTED** [review]

File: src/DipStrikeTool.hpp (hunk 1 above)
Comment (JonasGilg) [reply src/DipStrikeTool.hpp:63]: At this point it is already a private block.

File: src/PathTool.hpp (hunk 3 above)
Comment (JonasGilg) [reply src/PathTool.hpp:61]: Private was already declared.

File: src/PolygonTool.hpp (hunk 5 above)
Comment (JonasGilg) [reply src/PolygonTool.hpp:89]: Private was already declared.

File: src/PolygonTool.hpp (hunk 6 above)
Comment (JonasGilg) [reply src/PolygonTool.hpp:149]: I moved all back to being class members in case we introduce unity builds. This avoids name collisions.

**Review by Schneegans — APPROVED** [review]

**Comment by JonasGilg** [comment]
The ellipses now work the same as on the develop branch. On both branches the initial state looks wrong.

**Comment by Schneegans** [comment]
I fixed the initial ellipse state in https://github.com/cosmoscout/cosmoscout-vr/pull/138. You may merge this one here!

**Pull request merged**